manual/io/index.md

```diff
@@ -56,784 +46,433 @@ root [1] TFile f1("my.root");
 root [2] gDirectory->pwd()
 
 my.root:/
+root [3] hist->Draw() // for `hist` from `my.root`
 {% endhighlight %}
 
 `Rint:/` corresponds to "ROOT's in-memory" directory, the default directory during startup.
 
-## Viewing the contents of a ROOT file
-
-{% include ref class="TFile" %} is a descendent of {% include ref class="TDirectory" %}, which means it behaves like a `TDirectory`. You can list the contents, print the
-name, and create subdirectories. In a ROOT session, you are always in a directory and the directory you are in is
-called the current directory and is stored in the global variable `gDirectory` (see → [gDirectory]({{ '/manual/root_architecture_and_components/#gdirectory' | relative_url }})).
+### Adding and removing objects from a ROOT file
 
-### Physical layout of a ROOT file
+When writing an object to a ROOT file, ROOT creates a directory "entry" ({% include ref class="TKey" %}) representing the object, which mainly consists of a name and the object's persistent data.
+You can think of a `TFile` as a collection of `TKeys`, possibly inside nested `TDirectory`-s.
```

Comment:
`TDirectory`-s : the dash looks out of place.

Reply:
it's because `TDirectories` is weird and `TDirectory`s is wrong.

I'll go with `TDirectories` as the lesser evil.